src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -27,43 +25,60 @@ public DiagnosticServices(ILogger<DiagnosticsMonitor> logger)
 
         public IEnumerable<int> GetProcesses()
         {
-            //TODO This won't work properly with multi-container scenarios that don't share the process space.
-            //TODO We will need to use DiagnosticsAgent if we are the server.
-            return DiagnosticsClient.GetPublishedProcesses();
+            try
+            {
+                //TODO This won't work properly with multi-container scenarios that don't share the process space.
+                //TODO We will need to use DiagnosticsAgent if we are the server.
+                return DiagnosticsClient.GetPublishedProcesses();
+            }
+            catch (UnauthorizedAccessException)
+            {
+                throw new InvalidOperationException("Unable to enumerate processes.");
```

Comment:
>InvalidOperationException [](start = 26, length = 25)

I was wondering if we should define a custom exception across the service boundary. Doesn't have to be a part of this review but it may make it easier to communicate between the http rest layer and the service layer

Reply:
I had the same thought. I would at least make it easier to differentiate input errors vs service errors.